src/focus-visible.js

```diff
@@ -123,64 +126,34 @@ function init() {
    * @param {Event} e
    */
   function onBlur(e) {
-    if (e.target == document) {
+    if (e.target == document || e.target.nodeName == 'HTML') {
       return;
     }
 
-    removeFocusRingClass(e.target);
-  }
-
-  /**
-   * When the window regains focus, restore the focus-visible class to the element
-   * to which it was previously applied.
-   */
-  function onWindowFocus() {
-    // When removing the activeElement from DOM it's possible IE11 is in state
-    // document.activeElement === null
-    if (!document.activeElement) {
-      return;
+    if (e.target.classList.contains('focus-visible')) {
+      hadFocusVisibleRecently = true;
+      window.clearTimeout(hadFocusVisibleRecentlyTimeout);
+      hadFocusVisibleRecentlyTimeout = window.setTimeout(function() {
+        hadFocusVisibleRecently = false;
+        window.clearTimeout(hadFocusVisibleRecentlyTimeout);
+      }, 100);
+      removeFocusVisibleClass(e.target);
     }
-
-    if (document.activeElement == elWithFocusRing) {
-      addFocusRingClass(elWithFocusRing);
-    }
-
-    elWithFocusRing = null;
   }
 
-  /**
-   * When switching windows, keep track of the focused element if it has a
-   * focus-visible class.
-   * @param {Event} e
-   */
-  function onWindowBlur(e) {
-    if (e.target !== window) {
-      return;
-    }
-
-    // When removing the activeElement from DOM it's possible IE11 is in state
-    // document.activeElement === null
-    if (!document.activeElement) {
-      return;
-    }
-
-    // Set initial state back to assuming that the user is relying on the keyboard.
-    // And add listeners to detect if they use a pointing device instead.
-    hadKeyboardEvent = true;
-    addInitialPointerMoveListeners();
-
-    if (document.activeElement.classList.contains('focus-visible')) {
-      // Keep a reference to the element to which the focus-visible class is
-      // applied so the focus-visible class can be restored to it if the window
-      // regains focus after being blurred.
-      elWithFocusRing = document.activeElement;
+  function onVisibilityChange(e) {
+    if (document.visibilityState == 'hidden') {
+      if (hadFocusVisibleRecently) {
+        hadKeyboardEvent = true;
+      }
+      addInitialPointerMoveListeners();
     }
   }
 
   /**
-   * Add a group of listeners to detect usage of any pointing devices.
-   * These listeners will be added when the polyfill first loads, and anytime
-   * the window is blurred, so that they are active when the window regains focus.
+   * Add a group of listeners to detect usage of any pointing devices. These
```

Comment:
This somehow got reflowed. We should agree on a width so this doesn't keep happening.

Reply:
oh i think i used to have .vscode project settings which may have had it at 100 but those weren't checked into github so I reformated to a default 80